src/System.Management.Automation/engine/MshCommandRuntime.cs

```diff
@@ -3740,6 +3740,13 @@ internal void SetVariableListsInPipe()
 
```

Comment:
@vexx32, your last commit had 1 failures in `PowerShell-CI-windows`
Parameter Binding Tests.PipelineVariable shouldn't cause a NullRef exception when 'DynamicParam' block is present
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. Object reference not set to an instance of an object.
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\ParameterBinding\ParameterBinding.Tests.ps1: line 256
256:         DynamicParamTest -PipelineVariable bar | ForEach-Object { $bar } | Should -Be "hi"

```